### DISCUSSION TITLE
Using Dragonfly - question about BSL license?

### DISCUSSION THREAD

**Discussion by davejoyce-lab49** [dragonflydb/dragonfly#86] [Q&A] [answered]
As a consultant who develops distributed systems for clients, I am keenly interested in using Dragonfly as a component in my projects. Could you clarify which situations allow for open source licensed use vs commercial use, per the [BSL 1.1](https://github.com/dragonflydb/dragonfly/blob/main/LICENSE.md)?

The situations described in "Additional Use Grant" clause are unclear to me.

> Additional Use Grant: You may make use of the Licensed Work (i) only as part of your own product or service, provided it is not an in-memory data store product or service; and (ii) provided that you do not use, provide, distribute, or make available the Licensed Work as a Service. A “Service” is a commercial offering, product, hosted, or managed service, that allows third parties (other than your own employees and contractors acting on your behalf) to access and/or use the Licensed Work or a substantial set of the features or functionality of the Licensed Work to third parties as a software-as-a-service, platform-as-a-service, infrastructure-as-a-service or other similar services that compete with Licensor products or services.

If I build software for clients rather than for my firm to sell as a product, that requires a commercial license?

**Comment by romange** [comment, marked as answer]
The underlying intent is to provide protection against unfair competition.
If you build software that uses Dragonfly and this software has nothing to
do with in-memory stores or databases, i.e its primary goal is to fulfill a
business need that is not database related then you are clear.

While the license speaks in general terms, in practice, it protects us
against a handful companies and they wont come here to ask this question 😃

**Reply by davejoyce-lab49** [reply]
Thanks for the quick, succinct answer! 😃

**Comment by readall** [comment]
Although this is marked as answered. The text of BSL 1.1 has following statement:
"Text of BSL 1.1

The Licensor hereby grants you the right to copy, modify, create derivative works, redistribute, **and make non-production** use of the Licensed Work. The Licensor may make an Additional Use Grant, above, permitting limited production use. "

Does this mean I can not use dragonflydb source available version for having production grade setup, even if my product has noting to do with providing Database-as-service of any kind (in-memory or otherwise)?

Please help me understand.

**Reply by mainpart** [reply]
same question here. what about production?

**Reply by romange** [reply]
it's all the in the license text. Also we have FAQ.md in this repo that talks about this. TLDR: you can run dragonfly in production for free as long as you do not sell dragonfly APIs as a service.

```
Additional Use Grant: You may make use of the Licensed Work (i) only as part of your own product or service, provided it is not an in-memory data store product or service; and (ii) provided that you do not use, provide, distribute, or make available the Licensed Work as a Service. A “Service” is a commercial offering, product, hosted, or managed service, that allows third parties (other than your own employees and contractors acting on your behalf) to access and/or use the Licensed Work or a substantial set of the features or functionality of the Licensed Work to third parties as a software-as-a-service, platform-as-a-service, infrastructure-as-a-service or other similar services that compete with Licensor products or services.
```
